src/cpu-info.ts

```diff
@@ -1,18 +1,15 @@
-export type CpuInfo = {
-  numLogicalCores?: number;
-};
-
 /**
- * Get the available information about the machine's CPU.
- *
- * @returns The {@link CpuInfo}.
+ * Provides information about the CPU.
  */
-export function getCpuInfo(): CpuInfo {
-  const cpuInfo = {} as CpuInfo;
+export class CpuInfo {
+  private static numLogicalCores = navigator?.hardwareConcurrency;
```

Comment:
Is there any use of storing `numLogicalCores` as a property of `CpuInfo`? Could we instead just return `navigator.hardwareConcurrency` in `getNumLogicalCores` instead?